packages/grafana-ui/src/components/Combobox/Combobox.test.tsx

```diff
@@ -132,4 +132,73 @@ describe('Combobox', () => {
       expect(typeof onChangeHandler.mock.calls[1][0].value === 'number').toBeTruthy();
     });
   });
+
+  describe('async', () => {
+    beforeEach(() => {
+      jest.useFakeTimers();
+    });
+
+    // Assume that most apis only return with the value
+    const simpleAsyncOptions = [{ value: 'Option 1' }, { value: 'Option 2' }, { value: 'Option 3' }];
+
+    it('should allow async options', async () => {
+      const asyncOptions = jest.fn(() => Promise.resolve(simpleAsyncOptions));
+      render(<Combobox options={asyncOptions} value={null} onChange={onChangeHandler} />);
+
+      const input = screen.getByRole('combobox');
+      await userEvent.click(input, { delay: null });
+
+      expect(asyncOptions).toHaveBeenCalled();
+      //expect(screen.getByText('Loading')).toBeInTheDocument();
+    });
+
+    it('should allow async options and select value', async () => {
+      const asyncOptions = jest.fn(() => Promise.resolve(simpleAsyncOptions));
+      render(<Combobox options={asyncOptions} value={null} onChange={onChangeHandler} />);
+
+      const input = screen.getByRole('combobox');
+      await userEvent.click(input, { delay: null });
+
+      const item = await screen.findByRole('option', { name: 'Option 3' });
+      await userEvent.click(item, { delay: null });
+
+      expect(onChangeHandler).toHaveBeenCalledWith(simpleAsyncOptions[2]);
+      expect(screen.getByDisplayValue('Option 3')).toBeInTheDocument();
+    });
+
+    it('should ignore late responses', async () => {
```

Comment:
I can't get this scenario to fail when removing the if statement that should handle this case.